sherpa/astro/data.py

```diff
@@ -107,7 +107,7 @@ def __str__(self):
         old = self._fields
```

Comment:
By the way, something else I found out while reviewing this is that the base class's `__init__` is called without passing it any arguments apart from `self`. Then it looks like the base class inspects the frame to get the arguments passed to the calling class, which is weird. But again, I didn't look any closer.